### PR TITLE
Add riddle_sense dataset

### PR DESCRIPTION
**Pull request by ziyiwu9494**
Adding a new dataset for QA with riddles. I'm confused about the tagging process because it looks like the streamlit app loads data from the current repo, so is it something that should be done after merging or off my fork?

### CODE REVIEW

**Comment by ziyiwu9494** [comment]
@lhoestq 
I address all the comments, I think. Thanks!

File: datasets/riddle_sense/README.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+- monolingual
+pretty_name: RiddleSense
+size_categories:
+- unknown
+source_datasets:
+- original
+task_categories:
+- question-answering
+task_ids:
+- multiple-choice-qa
```
Comment (lhoestq) [inline datasets/riddle_sense/README.md:18]:
```suggestion
---
annotations_creators:
- crowdsourced
language_creators:
- found
languages:
- en
licenses:
- other-non-commercial
multilinguality:
- monolingual
pretty_name: RiddleSense
size_categories:
- 1K<n<10K
source_datasets:
- original
task_categories:
- question-answering
task_ids:
- multiple-choice-qa
---
```

File: datasets/riddle_sense/README.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+## Additional Information
+
+### Dataset Curators
+
+[Needs More Information]
+
+### Licensing Information
+
+[Needs More Information]
```
Comment (lhoestq) [inline datasets/riddle_sense/README.md:159]:
```suggestion
The copyright of RiddleSense dataset is consistent with the terms of use of the fan websites and the intellectual property and privacy rights of the original sources. All of our riddles and answers are from fan websites that can be accessed freely. The website owners state that you may print and download material from the sites solely for non-commercial use provided that we agree not to change or delete any copyright or proprietary notices from the materials. The dataset users must agree that they will only use the dataset for research purposes before they can access the both the riddles and our annotations. We do not vouch for the potential bias or fairness issue that might exist within the riddles. You do not have the right to redistribute them. Again, you must not use this dataset for any commercial purposes. 
```

File: datasets/riddle_sense/riddle_sense.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            description=_DESCRIPTION,
+            # datasets.features.FeatureConnectors
+            features=features,
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage="https://inklab.usc.edu/RiddleSense/",
+            citation=_CITATION,
```
Comment (lhoestq) [inline datasets/riddle_sense/riddle_sense.py:67]:
```suggestion
            citation=_CITATION,
            license=_LICENSE,
```

File: datasets/riddle_sense/riddle_sense.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+skills, which are all important abilities for advanced natural language understanding (NLU). However,
+there is currently no dedicated datasets aiming to test these abilities. Herein, we present RiddleSense,
+a new multiple-choice question answering task, which comes with the first large dataset (5.7k examples) for answering
+riddle-style commonsense questions. We systematically evaluate a wide range of models over the challenge,
+and point out that there is a large gap between the best-supervised model and human performance — suggesting
+intriguing future research in the direction of higher-order commonsense reasoning and linguistic creativity towards
+building advanced NLU systems.
+
+"""
+
```
Comment (lhoestq) [inline datasets/riddle_sense/riddle_sense.py:29]:
```suggestion

_LICENSE = """\
The copyright of RiddleSense dataset is consistent with the terms of use of the fan websites and the intellectual
property and privacy rights of the original sources. All of our riddles and answers are from fan websites that can be
accessed freely. The website owners state that you may print and download material from the sites solely for non
commercial use provided that we agree not to change or delete any copyright or proprietary notices from the
materials. The dataset users must agree that they will only use the dataset for research purposes before they can
access the both the riddles and our annotations. We do not vouch for the potential bias or fairness issue that might
exist within the riddles. You do not have the right to redistribute them. Again, you must not use this dataset for any
commercial purposes. 
"""

```

**Review by lhoestq — APPROVED** [review]
Nice thank you !

I also added mentions about the non-commercial usage of the dataset from their website

**Comment by lhoestq** [comment]
The five test fails are unrelated to this PR and fixed on master so we can ignore them

**Pull request merged**